src/ElectricPokemon.java

```diff
@@ -0,0 +1,46 @@
+import java.util.HashMap;
+
+public abstract class ElectricPokemon extends Pokemon{
+    private boolean hasCharge = false;
+    private boolean needsPowerstone = false;
+
+    /**
+     * @param name       Name of the pokemon
+     * @param trainer    Name of the trainer
+     * @param base_hp    The base HP of the pokemon
+     */
+    public ElectricPokemon(String name, String trainer, int base_hp) {
+        super(PokemonTypes.ELECTRIC, name, trainer, base_hp);
+
+        HashMap<String, Float> relativeStrength = new HashMap<String, Float>();
+        relativeStrength.put( PokemonTypes.GRAS  ,0.5f);
+        relativeStrength.put( PokemonTypes.WATER  ,1.5f);
+
+        overrideDmgFactors(relativeStrength);
+    }
+
+    @Override
+    public void baseAttack(Pokemon target){
+        attackMove(target, "tackle", PokemonTypes.NORMAL, 40);
+    }
+
+    public void thunder(Pokemon target){
+        attackMove(target, "thunder", PokemonTypes.ELECTRIC, 50);
+    }
+
+    public boolean isHasCharge() {
```

Comment:
Dit is een rare naam. Je kunt hier beter `hasCharge()` gebruiken, dat klinkt wat natuurlijker